Makefile.am

```diff
@@ -130,31 +130,24 @@ TESTS += exhaustive_tests
 endif
 
 ### Precomputed tables
-EXTRA_PROGRAMS = gen_ecmult_static_pre_g gen_ecmult_gen_static_prec_table
+EXTRA_PROGRAMS = precompute
 CLEANFILES = $(EXTRA_PROGRAMS)
 
-gen_ecmult_static_pre_g_SOURCES = src/gen_ecmult_static_pre_g.c
-gen_ecmult_static_pre_g_CPPFLAGS = $(SECP_INCLUDES)
-gen_ecmult_static_pre_g_LDADD = $(SECP_LIBS) $(COMMON_LIB)
-
-gen_ecmult_gen_static_prec_table_SOURCES = src/gen_ecmult_gen_static_prec_table.c
-gen_ecmult_gen_static_prec_table_CPPFLAGS = $(SECP_INCLUDES)
-gen_ecmult_gen_static_prec_table_LDADD = $(SECP_LIBS) $(COMMON_LIB)
+precompute_SOURCES = src/precompute.c
+precompute_CPPFLAGS = $(SECP_INCLUDES)
+precompute_LDADD = $(SECP_LIBS) $(COMMON_LIB)
 
 # See Automake manual, Section "Errors with distclean".
 # We don't list any dependencies for the prebuilt files here because
 # otherwise make's decision whether to rebuild them (even in the first
 # build by a normal user) depends on mtimes, and thus is very fragile.
 # This means that rebuilds of the prebuilt files always need to be
 # forced by deleting them, e.g., by invoking `make clean-precomp`.
-src/ecmult_static_pre_g.h:
-	$(MAKE) $(AM_MAKEFLAGS) gen_ecmult_static_pre_g$(EXEEXT)
-	./gen_ecmult_static_pre_g$(EXEEXT)
-src/ecmult_gen_static_prec_table.h:
-	$(MAKE) $(AM_MAKEFLAGS) gen_ecmult_gen_static_prec_table$(EXEEXT)
-	./gen_ecmult_gen_static_prec_table$(EXEEXT)
-
 PRECOMP = src/ecmult_gen_static_prec_table.h src/ecmult_static_pre_g.h
+$(PRECOMP)&:
```

Comment:
is the `&` on purpose? I've never seen this syntax before.

Reply:
Yes, it means: one command builds all these targets. Otherwise, if invoked with `-jN`, the submake commands gets invoked multiple times in parallel, which caused race condition failures on CI.

I guess I'll need another solution if mac doesn't support it.

Reply:
Ah that explanation helped me search the web: https://stackoverflow.com/questions/63928066/what-is-the-canonical-way-to-simulate-grouped-targets-in-older-versions-of-make
And this confirms that it does not work on Mac Make.

Reply:
I think you could make it a pattern rule, which have supported grouped targets for a long time, see last example here: https://www.gnu.org/software/make/manual/html_node/Pattern-Examples.html#Pattern-Examples

```
src/precomputed_%: 
	$(MAKE) $(AM_MAKEFLAGS) precompute$(EXEEXT)
	./precompute$(EXEEXT)
```

Reply:
How does the syntax in your suggested code indicate grouping?

Reply:
Oh, they always are. TIL

Reply:
> Pattern rules may have more than one target; however, every target must contain a % character. Pattern rules are always treated as grouped targets (see Multiple Targets in a Rule) regardless of whether they use the : or &: separator. 

I interpret that as that you actually need multiple patterns on the left of the column. Otherwise rules like `%.o: %.c` would not do what we expect them to do.

Reply:
Oh true... One could still abuse a pattern rule with % that essentially matches only once. But then not sure whether this is ugly than a stamp file.

Reply:
Hmm, what about just

```make
precomp:
	$(MAKE) $(AM_MAKEFLAGS) precompute$(EXEEXT)
	./precompute$(EXEEXT)
```
?

Reply:
Switched to a different approach, where `./precompute` is given the filename to generate, and it just generates that.